File: app/src/test/java/io/github/tobyhs/weatherweight/forecast/TestForecastModule.java
package io.github.tobyhs.weatherweight.forecast;

import dagger.Module;
import dagger.Provides;
import dagger.android.ContributesAndroidInjector;

import static org.mockito.Mockito.mock;

/**
 * Dagger module for {@link ForecastActivity} in tests
 */
@Module
public abstract class TestForecastModule {
    @ContributesAndroidInjector
    abstract ForecastActivity contributeForecastActivityInjector();

    /**
     * @return mock presenter
     */
    @Provides
    public static ForecastPresenter provideForecastPresenter() {
        return mock(ForecastPresenter.class);
    }
}

File: app/src/test/java/io/github/tobyhs/weatherweight/forecast/TestForecastModule.kt
package io.github.tobyhs.weatherweight.forecast

import dagger.Module
import dagger.Provides
import dagger.hilt.android.components.ActivityRetainedComponent
import dagger.hilt.testing.TestInstallIn

import org.mockito.Mockito.mock

/**
 * Dagger module for [ForecastActivity] in tests
 */
@Module
@TestInstallIn(components = [ActivityRetainedComponent::class], replaces = [ForecastModule::class])
class TestForecastModule {
    /**
     * @return mock presenter
     */
    @Provides
    fun provideForecastPresenter(): ForecastPresenter = mock(ForecastPresenter::class.java)
}
